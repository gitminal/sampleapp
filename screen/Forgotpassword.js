import { View, Text, StyleSheet, TextInput, StatusBar, TouchableOpacity } from 'react-native'
import React, { useState } from 'react'
import Icon from 'react-native-vector-icons/Ionicons';
import Colors from '../constant/Colors';
const Forgotpassword = ({ navigation }) => {

    const [Forgotpassword, setForgotPassword] = useState('');
    const [chForgotpassword, setchForgotpassword] = useState(true);
    const [ForgotpasswordValidError, setForgotpasswordValidError] = useState('');

    const handleForgotPassword = () => {
        let reg = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w\w+)+$/;
        if (Forgotpassword == "" || Forgotpassword == undefined || Forgotpassword == null) {
            setForgotpasswordValidError('email address must be enter');
            setchForgotpassword(false);
            return false;
        } else if (!reg.test(Forgotpassword)) {
            setForgotpasswordValidError('enter valid email address');
            setchForgotpassword(false);
            return false;
        } else {
            setForgotpasswordValidError("");
            setchForgotpassword(true);
            return true;
        }

    }
    return (
        <View style={styles.container}>
            <StatusBar barStyle="dark-content" backgroundColor="#ffff" />
            <View style={styles.logintext}>
                <Text style={styles.textlog}>Forgotpassword?</Text>
                <Text style={styles.subtext}>Welcome Home My Friend</Text>
            </View>

            <View style={styles.inputView}>
                <Icon name="mail-outline" size={15} style={styles.inputicon} />
                <TextInput
                    style={styles.TextInput}
                    placeholder="Email."
                    autoCorrect={false}
                    autoCapitalize="none"
                    onChangeText={setForgotPassword}
                    onEndEditing={handleForgotPassword}
                    testID="Forgot"
                />
            </View>
            <View>
                {chForgotpassword == true ? null : <Text style={{ color: "red" }}>{ForgotpasswordValidError}</Text>}
            </View>
            <TouchableOpacity style={styles.loginBtn} onPress={() => { if (handleForgotPassword()) { navigation.replace("Signup") } }}>
                <Text style={styles.loginbtntext}>SUBMIT</Text>
            </TouchableOpacity>
            <Text style={styles.account}>Already  have an account ?</Text>
            <View style={{ flexDirection: "row", alignItems: "center", justifyContent: "center" }}>
                <Text style={styles.textplease}>Please</Text>
                <TouchableOpacity onPress={() => { navigation.navigate("Login") }}>
                    <Text style={styles.textsignup} testID="Logintest"> Login</Text>
                </TouchableOpacity>
            </View>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: "#ffff",
        padding: 30
    },
    logintext: {
        marginTop: 30
    },
    textlog: {
        fontSize: 30,
        color: "black",
        marginTop: 20,
        fontWeight: "bold"
    },
    subtext: {
        fontSize: 15,
        marginTop: 20
    },
    inputView: {

        marginTop: 20,
        flexDirection: "row"
    },
    TextInput: {
        paddingLeft: 50,
        borderBottomWidth: 0.5,
        flex: 1,
        fontSize: 15
    },
    inputicon: {
        marginTop: 15,
        position: "absolute",
    },
    loginBtn:
    {
        borderRadius: 5,
        height: 50,
        alignItems: "center",
        justifyContent: "center",
        marginTop: 40,
        backgroundColor: Colors.primaryColor,
    },
    loginbtntext: {
        color: "white",
        fontSize: 20,
        fontWeight: "bold"
    },
    textforgot: {
        marginTop: 12,
    },
    account: {
        marginTop: 20,
        textAlign: 'center'
    },
    textsignup: {
        color: Colors.primaryColor
    },

})
export default Forgotpassword;
