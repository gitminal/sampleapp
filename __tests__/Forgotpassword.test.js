import React from "react";
import { render } from "@testing-library/react-native"
import Forgotpassword from "../screen/Forgotpassword";
import Adapter from "enzyme-adapter-react-16"
import enzyme from 'enzyme';
import { fireEvent, screen } from "react-native-testing-library";
import renderer from 'react-test-renderer';
enzyme.configure({ adapter: new Adapter() });

describe('<Forgotpass />', () => {
    test('renders correctly', () => {
        const rendered = renderer.create(<Forgotpassword />).toJSON();
        expect(rendered).toBeTruthy();
    });
    it('go to Login', () => {
        const navigation = { navigate: () => { } }
        spyOn(navigation, 'navigate');
        const page = render(<Forgotpassword navigation={navigation} />)
        const Login = page.getByTestId('Logintest');
        fireEvent.press(Login);
        expect(navigation.navigate).toHaveBeenCalledWith("Login");
    })
    it('render default ', () => {
        const { getAllByText, getByPlaceholderText,queryAllByText } = render(<Forgotpassword />)
        expect(getAllByText("SUBMIT").length).toBe(1);
        expect(getAllByText("Welcome Home My Friend").length).toBe(1);
        expect(getAllByText("Please").length).toBe(1);
        expect(getByPlaceholderText("Email"));
    })
    it('show Invalid Username and password', () => {
        const { getByTestId, getByText, queryAllByText, getAllByText } = render(<Forgotpassword />)
        fireEvent.changeText(getByTestId("Forgot"));
        queryAllByText('email address must be enter');
        expect(queryAllByText('Password  must be enter').length).toMatchSnapshot();
    })
})