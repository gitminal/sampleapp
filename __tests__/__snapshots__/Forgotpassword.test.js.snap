// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Forgotpass /> show Invalid Username and password 1`] = `0`;
